verifiersdemo/source/objc/EndpointOwnershipDemo/ViewController.m

```diff
@@ -0,0 +1,315 @@
+@property (weak, nonatomic) IBOutlet TWTRLogInButton *twtrLogInButton;
+@property (weak, nonatomic) IBOutlet GIDSignInButton *googleLogInButton;
+@property (weak, nonatomic) IBOutlet UIButton *logOutButton;
+
+@property (weak, nonatomic) IBOutlet UILabel *socialNetworkAuthorizationLabel;
+@property (weak, nonatomic) IBOutlet UILabel *kaaAuthorizationLabel;
+
+@property (weak, nonatomic) IBOutlet UIView *messagingView;
+@property (weak, nonatomic) IBOutlet UITextField *messageTextField;
+@property (weak, nonatomic) IBOutlet UITextView *chatTextView;
+
+@property (weak, nonatomic) IBOutlet UITextField *endpointAccessTokenTextField;
+@property (weak, nonatomic) IBOutlet UIView *attachView;
+
+@property (nonatomic, strong) KaaManager *kaaManager;
+@property (nonatomic, strong) User *user;
+@property (nonatomic, strong) EventFamilyFactory *eventFamilyFactory;
+@property (nonatomic, strong) VerifiersDemoEventClassFamily *vdecf;
+@property (nonatomic, strong) NSDateFormatter *chatDateFormatter;
+@property (nonatomic) AuthorizedNetwork authorizedNetwork;
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.chatDateFormatter = [[NSDateFormatter alloc] init];
+    [self.chatDateFormatter setDateFormat:@"HH:mm dd.MM.yy"];
+    
+    self.logOutButton.hidden = YES;
+    self.messagingView.hidden = YES;
+    self.attachView.hidden = YES;
+    self.googleLogInButton.colorScheme = kGIDSignInButtonColorSchemeDark;
+    
+    self.fbLoginButton.delegate = self;
+    [GIDSignIn sharedInstance].uiDelegate = self;
+    [GIDSignIn sharedInstance].delegate = self;
+    
+    self.twtrLogInButton.loginMethods = TWTRLoginMethodAll;
+    self.twtrLogInButton.logInCompletion = ^(TWTRSession *session, NSError *error) {
+        if (session) {
+            NSLog(@"Signed in as %@", [session userName]);
+            self.user = [[User alloc] initWithUserId:session.userID token:session.authToken authorizedNetwork:AuthorizedNetworkTwitter];
+            [self userHasLoggedIn];
+        } else {
+            NSLog(@"error: %@", [error localizedDescription]);
+        }
+    };
+    
+    self.kaaManager = [KaaManager sharedInstance];
+    [self.kaaManager startKaaClient];
+}
+
+- (void)userHasLoggedIn {
+    [self.kaaManager attachUser:self.user delegate:self];
+    [self updateAuthorizationStatusForLabel:AuthorizationLabelSocial status:YES];
+    
+    switch (self.user.network) {
+        case AuthorizedNetworkFacebook:
+            self.twtrLogInButton.hidden = YES;
+            self.googleLogInButton.hidden = YES;
+            break;
+            
+        case AuthorizedNetworkTwitter:
+        case AuthorizedNetworkGoogle:
+            self.fbLoginButton.hidden = YES;
+            self.twtrLogInButton.hidden = YES;
+            self.googleLogInButton.hidden = YES;
+            self.logOutButton.hidden = NO;
+            break;
+            
+        default:
+            break;
+    }
```

Comment:
Code duplication: almost the same part of code in userHasLoggedOut 
Refactor this 'switch' to the separate method that will take BOOL parameter.
This method will be used both in 'userHasLoggedIn' and 'userHasLoggedOut'